src/main/java/com/github/jreddit/parser/listing/SubmissionsListingParser.java

```diff
@@ -0,0 +1,56 @@
+package com.github.jreddit.parser.listing;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.jreddit.parser.entity.Submission;
+import com.github.jreddit.parser.entity.Thing;
+import com.github.jreddit.request.error.RedditError;
+
+public class SubmissionsListingParser extends RedditListingParser {
```

Comment:
I'm thinking these inherited classes of RedditListingParser could be ditched. Just add a generic type (i.e. RedditListingParser<T>) in base class then you get rid all of the duplication of code. 
The you'll just have RedditListingParser<Submission>, RedditListingParser<SubReddit> and so forth